### ISSUE TITLE
Question about answer+"#"

### ISSUE THREAD

**Issue by LIMANONDBannapol** [sarahESL/PubMedCLIP#10]
Hi, Thank you for sharing the wonderful work in the medical field. I have one question.

In QCR_PubMEDCLIP.lib.utils.create_label.py <line 200>

What is the reason for adding the "#" string to the answer file in an open answer? (for Slake dataset) 

`for answer in open_answers:
        if answer in ans2label:
            ans = answer + "#"`

**Comment by sarahESL** [comment]
So there are overlaps between the answers in open and close end types. E.g., `lung` is used as answer both for open-end questions (like `what is the organ in the image?`) and for closed-end questions (like `which organ in the image is bigger, lung or heart?`). This `#` was my initial attempt to distinguish these two cases.

**Comment by LIMANONDBannapol** [comment]
Thank you for your quick response. So, during training with the cross-entropy loss, the ground truth is the `one-hot vector `including the ans + "#"? For example, in the dataset, the candidate answer list only has 4 answers: `lung, lung#, MRI, MRI#`. If the ground truth of one question is `MRI#` from open-end (`MRI#`), the one-hot vector will be `[0, 0, 0, 1]`. Is this correct? If so, will this hurt the performance of the model because the model can also answer `MRI`? (MRI# and MRI are the same in semantics).

**Comment by dek924** [comment]
Hello, I have another question about this issue.

In your code, it handles open and close answers differently, but MEVF github [[github](https://github.com/aioz-ai/MICCAI19-MedVQA/blob/1868d906ec9b1f2dafac4315c097cba14941e2bc/tools/compute_softscore.py#L216)] doesn't seem to differentiate them.
Is it correct that the preprocessing to run the MEVF code should be done in the same way as the QCR model?
